engine/runtime/src/test/java/org/enso/interpreter/test/SignatureTest.java

```diff
@@ -602,6 +603,49 @@ public void suspendedArgumentsUnappliedFunction() throws Exception {
     }
   }
 
+  @Test
+  public void andConversions() throws Exception {
+    final URI uri = new URI("memory://and_conv.enso");
+    final Source src = Source.newBuilder("enso", """
+    from Standard.Base import all
+
+    type Plus
+        Impl value dict
+
+        + self (that:Plus) = if self.dict != that.dict then Panic.throw "panic!" else
+          self.dict.plus self.value that.value
+    type Mul
+        Impl value dict
+
+        * self (that:Mul) = if self.dict != that.dict then Panic.throw "panic!" else
+          self.dict.mul self.value that.value
+
+    compute (a : Plus & Mul) (b : Plus & Mul) =
+      add (x:Plus) (y:Plus) = x+y
+      p = add a b
+      m = a*b
+      add p m
+
+    type BooleanPlus
+        plus a:Boolean b:Boolean = a || b
+    Plus.from(that:Boolean) = Plus.Impl that BooleanPlus
+
```

Comment:
Implementation of `Plus` (above) and `Mul` (below) _type classes_ for `Boolean` via conversions.